ldap_auth_provider.py

```diff
@@ -265,6 +265,13 @@ def register_user(self, localpart, name, email_address):
         Returns:
             user_id (str): User ID of the newly registered user.
         """
+        if self.ldap_attributes["uid"] == self.ldap_attributes["mail"]:
```

Comment:
this changes the behaviour of `register_user` so that it no longer matches its docstring - please keep comments consistent with implementation!